app/controllers/api/v1/users_controller.rb

```diff
@@ -29,6 +29,48 @@ def update
         end
       end
 
+      # For social media logins, renders the appropriate `redirect_to` path depending on whether the user is registered.
+      #
+      # Requires that the ActionController::Parameters contain a :user key, with a nested :email key.
+      # For example: { user: { email: "john@example.com" } }
+      #
+      # @return [String] A string of the user's redirect_to path
+      # @see https://github.com/zquestz/omniauth-google-oauth2#devise
+      #
+      def exist
+        user = User.find_by(email: params[:user][:email])
+
+        @redirect_path = '/profile'
```

Comment:
Since we are not passing `@redirect_path` to a view, can we switch it to a local variable (i.e. `redirect_path`).

Reply:
done.